cdc/puller/sorter/file_backend.go

```diff
@@ -165,13 +214,20 @@ func (r *fileBackEndReader) readNext() (*model.PolymorphicEvent, error) {
 	if err != nil {
 		if err == io.EOF {
 			r.isEOF = true
+			// verifies that the file has not been truncated unexpectedly.
```

Comment:
Does the truncate here only happens by accident, not by cdc itself

Reply:
CDC will truncate a file once its content is no longer needed and the file is going to be recycled. Normally a file in use is not truncated. This piece of code here verifies against accidental truncation either by CDC bug or by user misoperations.